front/lib/api/auth0.ts

```diff
@@ -77,43 +105,40 @@ async function verifyAuth0Token(accessToken: string): Promise<jwt.JwtPayload> {
       },
       (err, decoded) => {
         if (err) {
-          reject(err);
-          return;
+          return resolve(new Err(err));
         }
         if (!decoded || typeof decoded !== "object") {
-          reject(new Error("No token payload"));
-          return;
+          return resolve(new Err(Error("No token payload")));
+        }
+
+        const payloadValidation = Auth0JwtPayloadSchema.decode(decoded);
+        if (isLeft(payloadValidation)) {
+          logger.error("Invalid token payload.");
+          return resolve(new Err(Error("Invalid token payload.")));
+        }
+
+        if (requiredScopes) {
```

Comment:
don't you need to check if the audience matches too ?

(and maybe skip the checks if not the right audience, as a temporary measure not to break the raycast extension)

Reply:
The audience is checked along with the expiration and issuer by jwt.verify

Reply:
I believe it's checked for consistency / signature but not for a specific value (may be wrong)

My point here is twofold:
- do the checks on scope only if the audience match the expected audience
- TEMPORARY, otherwise, let it go thru (so the raycast extension still works)

After the extension is updated, we'll wait a few days and then remove the temporary passthru.

Reply:
( jwt.verify checks that the audience matches the one passed in parameter )